src/libsyntax/attr.rs

```diff
@@ -312,6 +312,15 @@ pub fn find_export_name_attr(diag: &SpanHandler, attrs: &[Attribute]) -> Option<
     })
 }
 
+pub fn contains_extern_indicator(attrs: &[Attribute]) -> bool {
+    contains_name(attrs, "no_mangle") ||
+        contains_name(attrs, "export_name") ||
+        first_attr_value_str_by_name(attrs, "linkage").map(|v| match &*v {
```

Comment:
Linkage is indeed orthogonal, but it's actually the part that truly defines reachability. Changing the symbol from its default mangled name only implies an intent to have the symbol exist outside of Rust. `linkage` makes it explicit! And then there's also `link_section`, which I should probably add a check for as well...